code/modules/wiremod/core/integrated_circuit.dm

```diff
@@ -231,6 +253,11 @@ GLOBAL_LIST_EMPTY_TYPED(integrated_circuits, /obj/item/integrated_circuit)
 	if(!to_add.add_to(src))
 		return FALSE
 
+	if(to_add.circuit_flags & CIRCUIT_NO_DUPLICATES)
```

Comment:
@Watermelon914 Should this be in `add_component` or `add_component_manually`? I noticed that the print function does not use the manually proc? Is that intentional?

Reply:
I mean, this is the proc that encompasses it all.